webapp/app/[locale]/tunnel/_components/reviewOperation/chainIcon.tsx

```diff
@@ -0,0 +1,23 @@
+import { bitcoinMainnet, bitcoinTestnet } from 'btc-wallet/chains'
+import { BitcoinIcon } from 'components/reviewOperation/_icons/bitcoinIcon'
+import { EthIcon } from 'components/reviewOperation/_icons/ethIcon'
+import { HemiIcon } from 'components/reviewOperation/_icons/hemiIcon'
+import { hemi, hemiSepolia } from 'hemi-viem'
+import { RemoteChain } from 'types/chain'
+import { mainnet, sepolia } from 'viem/chains'
+
+interface Props {
+  chainId: RemoteChain['id']
+}
+
+const chainIconMap: Record<string | number, JSX.Element> = {
+  [sepolia.id]: <EthIcon />,
+  [mainnet.id]: <EthIcon />,
+  [hemiSepolia.id]: <HemiIcon />,
+  [hemi.id]: <HemiIcon />,
+  [bitcoinTestnet.id]: <BitcoinIcon />,
+  [bitcoinMainnet.id]: <BitcoinIcon />,
+}
+
+export const ChainIcon = ({ chainId }: Props): JSX.Element | null =>
+  chainIconMap[chainId] ?? null
```

Comment:
If `chainIconMap[chainId]` does not exist, it already returns `undefined`, which already renders nothing. So the `| null` adds no value and can be just removed.

I'd also use `ReactNode` in general for a "component rendered" instead of `JSX.Element`.  Lastly, if `chainIconMap` already specifies the return type, there's no need to then set it in the `ChainIcon`.

I'd also update the `Record` to use `RemoteChain['id']` to be consistent with the `Props`

So it'd go like this:


```suggestion
const chainIconMap: Record<RemoteChain['id'], ReactNode> = {
  [sepolia.id]: <EthIcon />,
  [mainnet.id]: <EthIcon />,
  [hemiSepolia.id]: <HemiIcon />,
  [hemi.id]: <HemiIcon />,
  [bitcoinTestnet.id]: <BitcoinIcon />,
  [bitcoinMainnet.id]: <BitcoinIcon />,
}

export const ChainIcon = ({ chainId }: Props) =>
  chainIconMap[chainId]
```

Reply:
Done.